diff --git a/CMakeLists.txt b/CMakeLists.txt
index 560bb403..3f5c7bbc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,8 @@
 # 3.9 is needed in 'doc' because of doxygen_add_docs()
 cmake_minimum_required(VERSION 3.5)
 
+cmake_policy(SET CMP0074 NEW)
+
 if(NOT (CMAKE_BUILD_TYPE OR CMAKE_CONFIGURATION_TYPES OR DEFINED ENV{CFLAGS} OR DEFINED ENV{CXXFLAGS}))
     set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo")
 endif()
diff --git a/include/FLAC++/export.h b/include/FLAC++/export.h
index d8dccf54..628bb3d7 100644
--- a/include/FLAC++/export.h
+++ b/include/FLAC++/export.h
@@ -56,7 +56,7 @@
  * \{
  */
 
-#if defined(FLAC__NO_DLL)
+#if 1 /*defined(FLAC__NO_DLL)*/
 #define FLACPP_API
 
 #elif defined(_WIN32)
diff --git a/include/FLAC/export.h b/include/FLAC/export.h
index 3e3e7648..79336c6e 100644
--- a/include/FLAC/export.h
+++ b/include/FLAC/export.h
@@ -56,7 +56,8 @@
  * \{
  */
 
-#if defined(FLAC__NO_DLL)
+/*#if defined(FLAC__NO_DLL)*/
+#if 1
 #define FLAC_API
 
 #elif defined(_WIN32)
